### PR TITLE
WIP: Implement cluster

### PR DESCRIPTION
**Pull request by BlackPoint-CX**
Implement cluster model based on MNIST with default parameters.
Result :
-----Training DEC Model-----
Using DataSource : mnist
With Parameters : 
	batch_size : 256
	maxiter : 8000
	update_interval : 140
	tol : 0.001
Getting Metrics : 
	acc : 0.55909
	nmi : 0.58148
	ari : 0.43699
Running Cost 1911.51 seconds.

### CODE REVIEW

**Comment by BlackPoint-CX** [comment]
This model is only used for verifying the training process with SQLFlow. NOT final version.
Some code like `ClusteringLayer` is based on work of [Tony607](https://github.com/Tony607/Keras_Deep_Clustering). 
I will re-implement it or add the quoting info in the final official version.

File: sqlflow_models/cluster.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    D = max(y_pred.max(), y_true.max()) + 1
+    w = np.zeros((D, D), dtype=np.int64)
+    for i in range(y_pred.size):
+        w[y_pred[i], y_true[i]] += 1
+
+    ind = linear_assignment(w.max() - w)
+    return sum([w[i, j] for i, j in ind]) * 1.0 / y_pred.size
+
+
+class DECModel(object):
```
Comment (tonyyang-svail) [inline sqlflow_models/cluster.py]: Should this model inherit from a `tf.keras.Model`?

**Review by tonyyang-svail — COMMENTED** [review]
Where are the tests? :)

File: sqlflow_models/cluster.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        self._n_clusters = n_clusters
+        self._activation_func = activation_func
+        self._batch_size = batch_size
+        self._pretrain_epochs = pretrain_epochs
+        self._kmeans_init = kmeans_init
+        self._pretrain_optimizer = SGD(lr=1, momentum=0.9)
+        self._optimizer = SGD(lr=0.01, momentum=0.9)
+        self._autoencoder, self._encoder = build_autoencoder(dims=self._encoder_dims,
+                                                             act=self._activation_func,
+                                                             init='glorot_uniform')
```
Comment (Yancey0623) [inline sqlflow_models/cluster.py]: Move the ClusteringLayer (L63) initialization func to be here?

**Comment by Yancey0623** [comment]
Hi @BlackPoint-CX , I found #16 did the same work, do you want to close this one and work on another one ?

**Pull request closed without merging**